client/client.go

```diff
@@ -319,9 +325,12 @@ func (h *httpSilenceAPI) Set(ctx context.Context, sil types.Silence) (string, er
 	var res struct {
 		SilenceID string `json:"silenceId"`
 	}
-	err = json.Unmarshal(body, &res)
 
-	return res.SilenceID, err
+	if err = json.Unmarshal(body, &res); err != nil {
+		return "", err
+	}
+
+	return res.SilenceID, nil
```

Comment:
Again, this seems needlessly wordy. I would only do it if there is any reason to ensure the string return is empty in case of an error. But I don't think there is. On the contrary.